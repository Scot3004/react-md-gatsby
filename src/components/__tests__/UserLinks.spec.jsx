import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserLinks from '../UserLinks';

const config = {
  userLinks: [
    {
      label: "GitHub",
      url: "https://github.com/Scot3004",
      iconClassName: "fa fa-github"
    },
    {
      label: "Facebook",
      url: "https://facebook.com/Scot3004",
      iconClassName: "fa fa-facebook"
    },
    {
      label: "Twitter",
      url: "https://twitter.com/Scot3004",
      iconClassName: "fa fa-twitter"
    }
  ],
}

it('renders correctly', () => {
  const tree = shallow(<UserLinks config={config} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = shallow(<UserLinks config={config} labeled />);
  expect(toJson(tree)).toMatchSnapshot();
});


it('div on empty userlinks', () => {
  const configEmpty = {userLinks: []}
  const tree = shallow(<UserLinks config={configEmpty} />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('null when not send userlinks', () => {
  const configNull = {}
  const tree = shallow(<UserLinks config={configNull} />);
  expect(toJson(tree)).toMatchSnapshot();
});
