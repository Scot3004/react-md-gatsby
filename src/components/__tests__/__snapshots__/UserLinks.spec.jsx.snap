// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div on empty userlinks 1`] = `
<div
  className="user-links"
/>
`;

exports[`null when not send userlinks 1`] = `""`;

exports[`renders correctly 1`] = `
<div
  className="user-links"
>
  <withInk(withTooltip(Button))
    href="https://github.com/Scot3004"
    icon={true}
    iconClassName="fa fa-github"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="GitHub"
    secondary={true}
  />
  <withInk(withTooltip(Button))
    href="https://facebook.com/Scot3004"
    icon={true}
    iconClassName="fa fa-facebook"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="Facebook"
    secondary={true}
  />
  <withInk(withTooltip(Button))
    href="https://twitter.com/Scot3004"
    icon={true}
    iconClassName="fa fa-twitter"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="Twitter"
    secondary={true}
  />
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="user-links"
>
  <withInk(withTooltip(Button))
    flat={true}
    href="https://github.com/Scot3004"
    icon={false}
    iconClassName="fa fa-github"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="GitHub"
    secondary={true}
  >
    GitHub
  </withInk(withTooltip(Button))>
  <withInk(withTooltip(Button))
    flat={true}
    href="https://facebook.com/Scot3004"
    icon={false}
    iconClassName="fa fa-facebook"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="Facebook"
    secondary={true}
  >
    Facebook
  </withInk(withTooltip(Button))>
  <withInk(withTooltip(Button))
    flat={true}
    href="https://twitter.com/Scot3004"
    icon={false}
    iconClassName="fa fa-twitter"
    inkTransitionEnterTimeout={450}
    inkTransitionLeaveTimeout={300}
    inkTransitionOverlap={150}
    key="Twitter"
    secondary={true}
  >
    Twitter
  </withInk(withTooltip(Button))>
</div>
`;
